MoneySplitter.Win10/Common/TransactionEventModelFactory.cs

```diff
@@ -0,0 +1,160 @@
+     
+        }
+
+        public IEnumerable<TransactionEventModel> GetFriendLends(IEnumerable<TransactionModel> friendTransactions, int friendId)
+        {
+
+            return GetTransactionEvents(friendTransactions.
+                Where(tr => tr.Owner.Id ==friendId), friendId);
+        }
+
+        public IEnumerable<TransactionEventModel> GetCommonTransationsByFriend(IEnumerable<TransactionModel> friendTransactions, int friendId)
+        {
+            return GetTransactionEvents(friendTransactions.
+                Where(tr => tr.Owner.Id != _membershipService.CurrentUser.Id && tr.Owner.Id != friendId));
+        }
+
+        public TransactionEventModel GetTransactionEvent(TransactionModel friendTransaction,
+			int? friendId = null,
+			bool isForceHide = false,
+			bool isDeadLineShowed = false)
+		{
+			return ConvertTransactionModelToTransactionEventModel(friendTransaction, friendId, isForceHide, isDeadLineShowed);
+		}
+
+		private UserRole GetUserRole(int userId, TransactionModel transaction)
+		{
+			if (transaction.Owner.Id == userId)
+			{
+				return UserRole.USER_TRANSACTION;
+			}
+			if (transaction.InProgressIds.Any(id => id == userId))
+			{
+				return UserRole.IN_PROGRESS;
+			}
+
+			if (transaction.FinishedIds.Any(id => id == userId))
+			{
+				return UserRole.FINISHED;
+			}
+
+			if (transaction.Collaborators.Any(cl => cl.Id == userId))
+			{
+				return UserRole.IN_BEGIN;
+			}
+			return UserRole.UNDEFINED;
+		}
+
+		private UserRole GetFriendRole(int? friendId, TransactionModel transactionModel)
+		{
+			if (friendId == null)
+			{
+				return UserRole.UNDEFINED;
+			}
+			return GetUserRole((int)friendId, transactionModel);
+		}
+		private int GetNotVisibilCollabarratorsCount(TransactionModel transactionModel)
+		{
+			var VisibilUserWithoutMeAndOwnerCount = transactionModel.Owner.Id == _membershipService.CurrentUser.Id ? 2 : 1;
+
+			return transactionModel.Collaborators.Count(cl => cl.Id != transactionModel.Owner.Id &&
+															  cl.Id != _membershipService.CurrentUser.Id) - VisibilUserWithoutMeAndOwnerCount;
+
+		}
+
+		private IEnumerable<string> GetVicibilCollaboratorsImageUrls(TransactionModel transactionModel)
+		{
+			const int MAX_COUNT_IMAGE = 4;
+			var OwnerAndMe = new List<string>();
+			OwnerAndMe.Add(transactionModel.Owner.ImageUrl);
+
+			if (transactionModel.Owner.Id != _membershipService.CurrentUser.Id)
+			{
+				OwnerAndMe.Add(_membershipService.CurrentUser.ImageUrl);
+			}
+
+			var allCollaborators = OwnerAndMe.Concat(transactionModel.Collaborators.
+														Where(cl => cl.Id != transactionModel.Owner.Id && cl.Id != _membershipService.CurrentUser.Id)
+														.Select(cl => cl.ImageUrl));
+
+			return allCollaborators.Count() <= MAX_COUNT_IMAGE ? allCollaborators : allCollaborators.Take(MAX_COUNT_IMAGE - 1);
+		}
+
+		private TypeDate GetTypeDate(TransactionModel transactionModel, bool IsDeadLineShowed)
+		{
+			if (IsDeadLineShowed ||
```

Comment:
symbols should be moved to second line,  I mean:
condition1
|| condition2
&& condition3....

Reply:
fixed